go/vt/vttablet/endtoend/message_test.go

```diff
@@ -461,3 +422,297 @@ func TestMessageAuto(t *testing.T) {
 		t.Errorf("message received:\n%v, want\n%v", got, want)
 	}
 }
+
+var createMessageTopic1 = `create table vitess_topic_subscriber_1(
+	time_scheduled bigint,
+	id bigint,
+	time_next bigint,
+	epoch bigint,
+	time_created bigint,
+	time_acked bigint,
+	message varchar(128),
+	primary key(time_scheduled, id),
+	unique index id_idx(id),
+	index next_idx(time_next, epoch))
+comment 'vitess_message,vt_topic=test_topic,vt_ack_wait=1,vt_purge_after=3,vt_batch_size=1,vt_cache_size=10,vt_poller_interval=1'`
+
+var createMessageTopic2 = `create table vitess_topic_subscriber_2(
+	time_scheduled bigint,
+	id bigint,
+	time_next bigint,
+	epoch bigint,
+	time_created bigint,
+	time_acked bigint,
+	message varchar(128),
+	primary key(time_scheduled, id),
+	unique index id_idx(id),
+	index next_idx(time_next, epoch))
+comment 'vitess_message,vt_topic=test_topic,vt_ack_wait=1,vt_purge_after=3,vt_batch_size=1,vt_cache_size=10,vt_poller_interval=1'`
+
+var createMessageTopic3 = `create table vitess_topic_subscriber_3(
+	time_scheduled bigint,
+	id bigint,
+	time_next bigint,
+	epoch bigint,
+	time_created bigint,
+	time_acked bigint,
+	message varchar(128),
+	primary key(time_scheduled, id),
+	unique index id_idx(id),
+	index next_idx(time_next, epoch))
+comment 'vitess_message,vt_topic=test_topic,vt_ack_wait=1,vt_purge_after=3,vt_batch_size=1,vt_cache_size=10,vt_poller_interval=1'`
+
+// TestMessageTopic tests for the case where id is an auto-inc column.
+func TestMessageTopic(t *testing.T) {
+	ch1 := make(chan *sqltypes.Result)
+	ch2 := make(chan *sqltypes.Result)
+	done := make(chan struct{})
+	client := framework.NewClient()
+
+	//
+	// phase 1 tests inserts into a topic going to two subscribed message tables
+	//
+	if _, err := client.Execute(createMessageTopic1, nil); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := client.Execute(createMessageTopic2, nil); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := client.Execute(createMessageTopic3, nil); err != nil {
```

Comment:
This test fails and deadlocks if I don't do something after making the first two tables that somehow flushes the table setup. Without this, the topic insert only runs on 1 of the 2 subscriber tables.